### PR TITLE
Fix several issues reported by static code analysis

### PR DESCRIPTION
**Pull request by stweil**
Signed-off-by: Stefan Weil <sw@weilnetz.de>

### CODE REVIEW

File: ugh/src/ugh/fileformats/excel/Excelfile.java [hunk 1]
```diff
@@ -2024,7 +2021,7 @@ public boolean CalculateEndPage(DocStruct inStruct)
 								// (physical struct).
 								List<Reference> refs = currentchild
 										.getAllReferences("to");
-								if (refs.size() == 0) {
+								if (refs == null || refs.size() == 0) {
 									// No references set, so set one to the
```
Comment (henning-gerhardt) [inline ugh/src/ugh/fileformats/excel/Excelfile.java]: Please put conditions in single round brackets:
`if ((refs == null) || (refs.size() == 0)) {`

Comment (stweil) [inline ugh/src/ugh/fileformats/excel/Excelfile.java]: IMHO that does not improve readability in this example. Is there a style guide which requires to use brackets here? Then much of the current code violates it. I found not a single line which uses the pattern which you suggest.

**Comment by stweil** [comment]
@henning-gerhardt, I did not find a style guide in the wiki. Could you please help me with a URL pointing to it?

**Comment by henning-gerhardt** [comment]
Coding guidelines are available under following link: http://www.kitodo.org/fileadmin/groups/kitodo/Dokumente/Kitodo-EntwicklerLeitfaden2016.pdf (nothing new, should already well known).

File: ugh/src/ugh/fileformats/excel/Excelfile.java (hunk 1 above)
Comment (henning-gerhardt) [inline ugh/src/ugh/fileformats/excel/Excelfile.java]: Maybe this rule is not written down in coding guideline but used in Kitodo.Production for new code.

**Comment by stweil** [comment]
There exists a German documentation called [Leitfaden für Kitodo-Entwickler](https://www.kitodo.org/fileadmin/groups/kitodo/Dokumente/Kitodo-EntwicklerLeitfaden2016.pdf) (guide for Kitodo developers).

It does not contain a rule how to write Java conditionals.

It contains some rules for Kitodo.Production. According to these rules the [AmbySoft Inc. Coding Standards for Java](http://www.ambysoft.com/downloads/javaCodingStandards.pdf) are required. These standards do not require parentheses for simple conditionals like the one in this PR. They suggest using parentheses for more complex conditionals. That is fine. They also suggest putting constants on the left side (`null == refs`). That's a bad idea. So even that standards should only be used [cum grano salis](https://de.wikipedia.org/wiki/Cum_grano_salis).

For Kitodo.UGH there are currently no explicit rules. Therefore I suggest sticking to existing conventions for that project.

**Comment by henning-gerhardt** [comment]
Kitodo.UGH and Kitodo.ContentServer are IMHO parts of Kitodo.Production, so rules of Kitodo.Production apply here too. Must everything written down? Coding style should be a readable style and this include for me round brackets around every condition.

I will accept this PR this time and for similar changes.

**Pull request merged**